src/tracks/ui/BrushHandle.cpp

```diff
@@ -282,31 +285,44 @@ UIHandle::Result BrushHandle::Drag
    mMostRecentX = x;
    mMostRecentY = y;
 
+   double posTime;
+   wxInt64 posFreq;
+   auto posToLongLong = [&](int x0){
+      posTime = viewInfo.PositionToTime(x0, mRect.x);
+      sampleCount sc = wt->TimeToLongSamples(posTime);
+      return sc.as_long_long();
+   };
+
    // Clip the coordinates
    // TODO: Find ways to access the ClipParameters (for the mid)
-   int x1 = std::max(mRect.x + 10, std::min(x, mRect.x + mRect.width - 20));
-   int y1 = std::max(mRect.y + 10, std::min(y, mRect.y + mRect.height - 10));
+   int dest_xcoord = std::max(mRect.x + 10, std::min(x, mRect.x + mRect.width - 20));
+   int dest_ycoord = std::max(mRect.y + 10, std::min(y, mRect.y + mRect.height - 10));
+   int destFreq = PositionToFrequency(wt, 0, dest_ycoord, mRect.y, mRect.height);
+
+   int x1 = posToLongLong(dest_xcoord) / mpSpectralData->GetHopSize();
+   int y1 = destFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
 
    mbCtrlDown = event.ControlDown();
 
+   // Use the hop and bin number to calculate the brush stroke, instead of the mouse coordinates
+   // For mouse coordinate:
+   // (src_xcoord, src_ycoord) -> (dest_xcoord, dest_ycoord)
+   //
+   // Variables for Bresenham's line algorithm:
+   // x0, x1, y0, y1... (already transformed in hops and bins space}
    if(!mpSpectralData->coordHistory.empty()){
-      int x0 = mpSpectralData->coordHistory.back().first;
-      int y0 = mpSpectralData->coordHistory.back().second;
+      int src_xcoord = mpSpectralData->coordHistory.back().first;
+      int src_ycoord = mpSpectralData->coordHistory.back().second;
+      int srcFreq = PositionToFrequency(wt, 0, src_ycoord, mRect.y, mRect.height);
+
+      int x0 = posToLongLong(src_xcoord) / mpSpectralData->GetHopSize();
+      int y0 = srcFreq / (mpSpectralData->GetSR() / mpSpectralData->GetWindowSize());
       int wd = mBrushRadius * 2;
 
       int dx =  abs(x1-x0), sx = x0<x1 ? 1 : -1;
       int dy = -abs(y1-y0), sy = y0<y1 ? 1 : -1;
       int err = dx+dy, err2;
```

Comment:
"Line drawing" steps are unchanged.  Only the coordinate space is different.  That's what I expected.  (But maybe you will change variable names.  Also dh, db)